src/main/java/org/mitre/synthea/export/TextExporter.java

```diff
@@ -242,8 +401,33 @@ private static void basicInfo(List<String> textRecord, Person person, long endTi
    */
   private static void encounter(List<String> textRecord, Encounter encounter) {
     String encounterTime = dateFromTimestamp(encounter.start);
+    if (encounter.reason == null){
+      textRecord.add(encounterTime + " : " + encounter.codes.get(0).display);
+    } else if (encounter.reason == null && encounter.provider != null) {
+      textRecord.add(encounterTime + " : Encounter at " + encounter.provider.name);
+    } else if (encounter.reason != null && encounter.provider == null) {
+      textRecord.add(encounterTime + " : Encounter for " + encounter.reason.display);
+    } else {
+      textRecord.add(encounterTime + " : Encounter at " + encounter.provider.name
+          + " : Encounter for " + encounter.reason.display);
+    }
+    textRecord.add("   ");
+  } 
+
+  /**
+   * Add all info from the encounter to the record.
+   *
+   * @param textRecord
+   *          Text format record, as a list of lines
+   * @param person
+   *          The person to export
+   * @param encounter
+   *          The encounter all of the information refers to
+   */
+  private static void encounterReport(List<String> textRecord, Person person, Encounter encounter) {
+    String encounterTime = dateFromTimestamp(encounter.start);
 
-    if (encounter.reason == null && encounter.provider == null) {
+    if (encounter.reason == null){
```

Comment:
Looks like this `&& encounter.provider == null)` got lost in the rebase. This should still be in there.